### PR TITLE
MAYA-128473 - Move UFE to its own Project - Part11 (UIInfoHandler)

### PR DESCRIPTION
**Pull request by seando-adsk**
#### MAYA-128473 - Move UFE to its own Project - Part11 (UIInfoHandler)
* Moved UsdUIInfoHandler to UsdUfe and split off Maya version.
* Moved Outliner (treeView) icons to UsdUfe (except MayaReference).

### CODE REVIEW

File: lib/mayaUsd/ufe/MayaUsdUIInfoHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+MayaUsdUIInfoHandler::MayaUsdUIInfoHandler()
+    : UsdUfe::UsdUIInfoHandler()
+{
+    // Register a callback to invalidate the invisible color.
+    fColorChangedCallbackId = MEventMessage::addEventCallback(
+        "DisplayRGBColorChanged", onColorChanged, reinterpret_cast<void*>(this));
```
Comment (seando-adsk) [inline lib/mayaUsd/ufe/MayaUsdUIInfoHandler.cpp:29]: Split off Maya stuff into this derived class. Mostly to handle this Maya callback when the color prefs changed.

File: lib/mayaUsd/ufe/MayaUsdUIInfoHandler.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    auto supportedTypes = Parent::getSupportedIconTypes();
+
+    // We support these node types directly.
+    static const UsdUfe::UsdUIInfoHandler::SupportedTypesMap mayaSupportedTypes {
+        { "MayaReference", "out_USD_MayaReference.png" },
+        { "ALMayaReference", "out_USD_MayaReference.png" }, // Same as mayaRef
+    };
+    supportedTypes.insert(mayaSupportedTypes.begin(), mayaSupportedTypes.end());
+    return supportedTypes;
+}
```
Comment (seando-adsk) [inline lib/mayaUsd/ufe/MayaUsdUIInfoHandler.cpp:89]: Overridden method to add the two Maya reference icon types.

File: lib/usdUfe/resources/icons/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+foreach(ICON_BASE ${TREEVIEW_ICONS})
+    # The _100.png files need to be installed without the _100. This is the
+    # base icon name that is used. A DCC (such as Maya) will automatically
+    # choose the _150/_200 image if neeeded.
+    install(FILES "out_USD_${ICON_BASE}_100.png"
+        DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/icons"
+        RENAME "out_USD_${ICON_BASE}.png"
+    )
+    install(FILES "out_USD_${ICON_BASE}_150.png" "out_USD_${ICON_BASE}_200.png"
+        DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/icons"
```
Comment (seando-adsk) [inline lib/usdUfe/resources/icons/CMakeLists.txt:47]: Install most of the TreeView (Outliner) icons from UsdUfe. After discussing with Julien I kept the original naming "out_USD_XXX.png" (with the 150/200 versions).

File: lib/usdUfe/ufe/UsdUIInfoHandler.cpp [hunk 4]
```diff
@@ -159,15 +114,10 @@ bool UsdUIInfoHandler::treeViewCellInfo(const Ufe::SceneItem::Ptr& item, Ufe::Ce
     return changed;
 }
 
-Ufe::UIInfoHandler::Icon UsdUIInfoHandler::treeViewIcon(const Ufe::SceneItem::Ptr& item) const
+UsdUIInfoHandler::SupportedTypesMap UsdUIInfoHandler::getSupportedIconTypes() const
```
Comment (seando-adsk) [inline lib/usdUfe/ufe/UsdUIInfoHandler.cpp:117]: New virtual method to return the list of supported icon types. Can be overridden by derived class to add extra icons (such as from the MayaUsdUIInfoHandler).

File: lib/usdUfe/ufe/UsdUIInfoHandler.h [hunk 5]
```diff
@@ -51,18 +51,19 @@ class MAYAUSD_CORE_PUBLIC UsdUIInfoHandler : public Ufe::UIInfoHandler
...
-    //       to be compatible with MMessage callback API.
-    static void onColorChanged(void*);
+    // Called from treeViewIcon() method to find the correct icon based on node type.
+    // Get the map (node type -> icon filename) of the supported icon types.
+    // Can be overridden by derived class to add to the map.
+    typedef std::map<std::string, std::string> SupportedTypesMap;
+    virtual SupportedTypesMap                  getSupportedIconTypes() const;
 
+protected:
+    // Derived classes can set this color to override the default invisible color.
```
Comment (seando-adsk) [inline lib/usdUfe/ufe/UsdUIInfoHandler.h:63]: Made fInvisibleColor protected so it can be modified by derived class (as it is in MayaUsdUIInfoHandler).

**Review by seando-adsk — COMMENTED** [review]
It looks bad with 103 files, but most of those are just moving of icons. The actual number of files to review is much less and changes are not that big.

**Review by ppt-adsk — APPROVED** [review]

**Review by deboisj — APPROVED** [review]

**Pull request merged**